compiler/rustc_resolve/src/imports.rs

```diff
@@ -699,6 +699,27 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                         }
                     }
                 }
+
+                if let NameBindingKind::Import { import, .. } = binding.kind
+                    && let Some(binding_id) = import.id()
+                    && let import_def_id = self.local_def_id(binding_id)
+                    && self.effective_visibilities.is_exported(import_def_id)
+                    && let Res::Def(reexported_kind, reexported_def_id) = binding.res()
+                    && !matches!(reexported_kind, DefKind::Ctor(..))
+                    && !reexported_def_id.is_local()
+                    && self.tcx.is_private_dep(reexported_def_id.krate)
```

Comment:
Nit: it makes sense to put the expensive checks last (`effective_visibilities.is_exported`, then `is_private_dep`) and cheap but effective checks first (`reexported_def_id.is_local()`).

Reply:
Or maybe this is fine, the `is_exported` check will reject 90+% of imports right away.

Reply:
No worries, now it should be as it was.